src/views/ModelPlan/TaskList/ITSolutions/SolutionImplementation/index.tsx

```diff
@@ -319,158 +384,57 @@ const SolutionImplementation = () => {
-                                            label={translateOpNeedsStatusType(
-                                              key
-                                            )}
-                                            value={key}
-                                            checked={
-                                              values.solutions[index]
-                                                ?.status === key
-                                            }
-                                            onChange={() => {
-                                              setFieldValue(
-                                                `solutions[${index}].status`,
-                                                key
-                                              );
-                                            }}
-                                          />
-                                        ))}
-                                      </Fieldset>
-                                    </FieldGroup>
-
-                                    <ImplementationStatuses />
-                                  </>
-                                )}
-
-                                {index !==
-                                  operationalNeed.solutions.length - 1 && (
-                                  <Divider className="margin-bottom-6 margin-top-6" />
-                                )}
-                              </div>
-                            );
-                          })}
+                          {isUpdatingStatus
+                            ? operationalNeed.solutions
+                                .filter(solution => solution.id === solutionId)
+                                .map((solution, index) => {
+                                  const identifier = (
+                                    solution.nameOther ||
+                                    solution.key ||
+                                    ''
+                                  ).replaceAll(' ', '-');
+
+                                  return (
+                                    <Solution
+                                      key={solution.id}
+                                      solution={solution}
+                                      identifier={identifier}
+                                      index={index}
+                                      handleOnBlur={handleOnBlur}
+                                      setFieldValue={setFieldValue}
+                                      isUpdatingStatus={isUpdatingStatus}
+                                      operationalNeed={operationalNeed}
+                                      values={values}
+                                      flatErrors={flatErrors}
+                                      loading={loading}
+                                    />
+                                  );
+                                })
```

Comment:
This conditional is a little redundant.  You should map over this only once.  The solutions to render should all be in an array.  If updating one, it should map over [solution].  If over all the solutions, map over them normally.   If this doesn't make sense lets pair!

This filtering needs to happen outside of the formik context, because on the submission, it is updating all the solutions, when we are only updating one.  It is having unwanted mutations on solutions that arent being updated.